File: packages/node/src/labels/type.ts
export class Label {
    private name: string;
    private value: string;
    constructor(name: string, value: string) {
        this.name = name;
        this.value = value;
    }

    public getName(): string {
        return this.name;
    }

    public getValue(): string {
        return this.value;
    }

    static Name(name: string) : (value: string) => Label {
        return (value: string) => new Label(name, value);
    }
}

File: packages/go/labels/type.go
package labels

type Label struct {
	name  string
	value string
}

func NewLabel(name string, value string) Label {
	return Label{name: name, value: value}
}

func Name(name string) func(string) Label {
	return func(value string) Label {
		return NewLabel(name, value)
	}
}

func (l Label) Name() string {
	return l.name
}

func (l Label) Value() string {
	return l.value
}
